fe1-web/src/features/witness/network/WitnessStoreWatcher.ts

```diff
@@ -152,21 +157,31 @@ export const makeWitnessStoreWatcher = (
       return;
     }
 
-    const messagesToWitness = currentAllIds.filter(
+    let messagesToWitness = currentAllIds.filter(
       (msgId) =>
         !currentUnprocessedIds.includes(msgId) &&
         (!previousAllIds.includes(msgId) || previousUnprocessedIds.includes(msgId)),
     );
```

Comment:
This is also a possible source of performance issues as it is $O(n^2)$, maybe we can improve this at a later point :)